pages/trial.tsx

```diff
@@ -0,0 +1,372 @@
+import { isBrowser } from "../utils/isBrowser";
+
+interface FormError {
+  email: string;
+  generic: string;
+  subdomain: string;
+}
+
+export default function Trial({ props }) {
+  const {
+    handleSubmit,
+    register,
+    formState: { errors },
+  } = useForm();
+  const [loading, setLoading] = useState(false);
+  const [registered, setRegistered] = useState(false);
+  const [editSubdomain, setEditSubdomain] = useState(false);
+  const [subdomain, setSubdomain] = useState("");
+  const [error, setError] = useState<FormError>({
+    email: null,
+    generic: null,
+    subdomain: null,
+  });
+  const urlPattern =
+    /[-a-zA-Z0-9@:%_\+.~#?&//=]{2,256}\.[a-z]{2,4}\b(\/[-a-zA-Z0-9@:%_\+.~#?&//=]*)?/gi;
+
+  const onSubdomainChange = (e) => {
+    e.preventDefault();
+    let extractedDomain: string = "";
+    if (e.target.name === "companyWebsite") {
+      extractedDomain = extractDomain(encodeURIComponent(e.target.value));
+      if (extractedDomain.includes(".")) {
+        let toTrim = extractedDomain.indexOf(".");
+        extractedDomain = extractedDomain.substring(0, toTrim);
+      }
+    } else {
+      extractedDomain = e.target.value;
+    }
+    setSubdomain(extractedDomain);
+  };
+
+  const onEditSubdomainClick = (e) => {
+    setEditSubdomain(!editSubdomain);
+  };
+
+  const errorHandler = new ErrorHandler();
+
+  errorHandler.subscribe("result", (e) => {
+    const error: FormError = { email: null, generic: null, subdomain: null };
+    const message = e?.error?.message || e?.message || e.toString();
+    console.log("error", message);
+    if (message.match(/email/i)) {
+      error.email = "Invalid email address.";
+    } else if (message.includes("subdomain")) {
+      error.subdomain = message;
+    } else {
+      error.generic = message;
+    }
+    setError(error);
+  });
+
+  const { execApi } = useApi(errorHandler);
+
+  const onSubmit = async (data) => {
+    console.log(data);
+    setLoading(true);
+    console.log(errors);
+    // defaults
+    let source = "/trial";
+    let medium = "web";
+    let campaign = globalThis?.document?.referrer;
+
+    const storage = globalThis?.sessionStorage;
+    if (storage) {
+      source = storage.getItem("utm_source")
+        ? storage.getItem("utm_source")
+        : storage.getItem("prevPath")
+        ? storage.getItem("prevPath")
+        : source;
+      medium = storage.getItem("utm_medium")
+        ? storage.getItem("utm_medium")
+        : medium;
+      campaign = storage.getItem("utm_campaign")
+        ? storage.getItem("utm_campaign")
+        : campaign;
+    }
+
+    data.subdomain = subdomain;
+    data.source = source;
+    data.medium = medium;
+    data.campaign = campaign;
+
+    // TO DO: CONFIGURE GOOGLE ANALYTICS CONVERSION DATA
+    if (isBrowser() && globalThis?.gtag) {
+      globalThis.gtag("event", "conversion", {
+        send_to: "send_to': 'AW-467110449/dWjCCL2x78ICELGU3t4B", // demoRequest conversion
```

Comment:
This is likely a new adWords conversion event, and should have a new ID (cc @andyjih to make the new event ID)

Reply:
@evantahler That is the new ID. I think the comments are just out of date.

Reply:
Oops!  Yes I pasted in the new id and did not update the comment.